packages/react-table/src/components/TableComposable/Td.tsx

```diff
@@ -0,0 +1,195 @@
+  actionsDropdownDirection,
+  isExpanded = null,
+  isCompoundExpanded = null,
+  onCollapse,
+  noPadding,
+  width,
+  visibility,
+  innerRef,
+  ...props
+}: TdProps) => {
+  const selectParams = onSelect
+    ? selectable(children as IFormatterValueType, {
+        columnIndex,
+        rowIndex,
+        rowData: {
+          selected: isSelected,
+          disableSelection
+        },
+        column: {
+          extraParams: {
+            onSelect: onSelect as OnSelect,
+            selectVariant: selectVariant as 'checkbox' | 'radio'
+          }
+        }
+      })
+    : null;
+  const actionParamsFunc = actions ? cellActions(actions, null, null) : null;
+  const actionParams = actionParamsFunc
+    ? actionParamsFunc(null, {
+        rowIndex,
+        columnIndex,
+        rowData: {
+          disableActions: actionsDisabled
+        },
+        column: {
+          extraParams: {
+            dropdownPosition: actionsDropdownPosition,
+            dropdownDirection: actionsDropdownDirection
+          }
+        }
+      })
+    : null;
+  const expandableParams =
+    isExpanded !== null
+      ? collapsible(null, {
+          rowIndex,
+          columnIndex,
+          rowData: {
+            isOpen: isExpanded
+          },
+          column: {
+            extraParams: {
+              onCollapse
+            }
+          }
+        })
+      : null;
+  const compoundParams =
+    isCompoundExpanded !== null
+      ? compoundExpand(
+          {
+            title: children,
+            props: {
+              isOpen: isCompoundExpanded
+            }
+          } as IFormatterValueType,
+          {
+            columnIndex,
+            rowIndex,
+            column: {
+              extraParams: {}
+            }
+          }
+        )
+      : null;
+  const widthParams = width ? cellWidth(width)() : null;
+  const visibilityParams = visibility
+    ? classNames(...visibility.map((vis: keyof IVisibility) => Visibility[vis]))()
+    : null;
+  const Component: any = (selectParams && selectParams.component) || component;
+  const transformedChildren =
+    (selectParams && selectParams.children) ||
+    (actionParams && actionParams.children) ||
+    (expandableParams && expandableParams.children) ||
+    (compoundParams && compoundParams.children) ||
+    children;
+  return (
+    <Component
+      data-label={dataLabel}
+      data-key={columnIndex}
+      className={css(
+        className,
+        textCenter && styles.modifiers.center,
+        noPadding && styles.modifiers.noPadding,
+        selectParams && selectParams.className,
+        actionParams && actionParams.className,
+        expandableParams && expandableParams.className,
+        compoundParams && compoundParams.className,
+        modifier && styles.modifiers[modifier as 'breakWord' | 'fitContent' | 'nowrap' | 'truncate' | 'wrap'],
+        widthParams && widthParams.className,
+        visibilityParams && visibilityParams.className
```

Comment:
Can be simplified
```suggestion
        widthParams?.className,
        visibilityParams?.className
```

Reply:
I've made some changes in the code to use the pre-existing `mergeProps` function which simplified this area